### PR TITLE
Move /deploy to /manifests, update RBAC rules

### PR DESCRIPTION
**Pull request by benjaminapetersen**
- Moves `/deploy` to `/manifests`
- Leaves `/deploy/cr.yaml` in `/deploy` as it ought not be automatically created
- Updates RBAC to use specific/correct rules

@enj

### CODE REVIEW

File: manifests/02-operator.yaml [hunk 1]
```diff
@@ -29,4 +29,4 @@ spec:
             fieldRef:
               fieldPath: metadata.namespace
         - name: OPERATOR_NAME
-          value: "console-operator"
+          value: "console-operator"
```
Comment (enj) [inline manifests/02-operator.yaml]: Looks accidental.

File: manifests/00-rbac-role.yaml [hunk 2]
```diff
@@ -1,40 +1,59 @@
...
   - console.openshift.io
   resources:
-  - "*"
+  - consoles
   verbs:
-  - "*"
+  - get
+  - list
+  - watch
+  - create
```
Comment (enj) [inline manifests/00-rbac-role.yaml]: Why?

File: manifests/00-rbac-role.yaml [hunk 3]
```diff
@@ -1,40 +1,59 @@
...
+  - update
+  - delete
```
Comment (enj) [inline manifests/00-rbac-role.yaml]: Add TODO to drop all the deletes

File: manifests/00-rbac-role.yaml (hunk 2 above)
Comment (benjaminapetersen) [reply manifests/00-rbac-role.yaml]: Why....? Whats the full question.

File: manifests/00-rbac-role.yaml (hunk 3 above)
Comment (benjaminapetersen) [reply manifests/00-rbac-role.yaml]: added (above, line 7, as it applies to many)

**Pull request merged**